startups/views.py

```diff
@@ -207,37 +203,55 @@ def create(self, request):
             return Response(data, status=status.HTTP_201_CREATED)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
-    # def update(self, request, pk=None):
-    #     # Implementation of PUT METHOD for one project - ExampLE URL: /api/projects/2/
-    #     # Do not forget about SLASH at the end of URL
-    #     # + you should send data in JSON
-    #     project_id = pk
-    #     project_updated_info = request.data
-    #     # ...
-    #     # PUT logic
-    #     # ...
-    #     data = {
-    #         'project_id': project_id,
-    #         'message': f"Hello, here's a PUT method! You update ALL information about PROJECT № {project_id}",
-    #         'updated_data': project_updated_info,
-    #         'status': status.HTTP_200_OK
-    #     }
-    #     return Response(data, status=status.HTTP_200_OK)
-    #
-    # def partial_update(self, request, pk=None):
-    #     # Implementation of PATCH METHOD for one project - ExampLE URL: /api/projects/2/
-    #     # Do not forget about SLASH at the end of URL
-    #     # + you should send data in JSON
-    #     # PATCHcing logic
-    #     project_id = pk
-    #     project_specific_updated_info = request.data
-    #     data = {
-    #         'project_id': project_id,
-    #         'message': f"Hello, here's a PATCH method! You update SOME information about project № {project_id}",
-    #         'specific_updated_data': project_specific_updated_info,
-    #         'status': status.HTTP_200_OK
-    #     }
-    #     return Response(data, status=status.HTTP_200_OK)
+    def update(self, request, pk=None):
+        """
+        Update an existing project with all fields required
+        (PUT api/projects/<pk>/)
+        Do not forget about SLASH at the end of URL
+        """
+        if not pk:
+            return Response({"error": "Method PUT not allowed"}, status=status.HTTP_400_BAD_REQUEST)
+        try:
+            instance = Project.objects.get(pk=pk)
```

Comment:
Rename `instance` here to 'project'
`project= Project.objects.get(pk=pk)`

Because previously in Startups, you used:

```
- startup = Startup.objects.filter(owner=request.user).first()
- startup = Startup.objects.filter(id=startup_id).first()
- startup = Startup.objects.filter(owner=user).first()
```

Reply:
це метод Олександра! стосується проєктів

Reply:
Ок - виправлю у себе просто.

Reply:
what does "instance" mean? it looks bad. We will fix it after Friday.

Reply:
this class ProjectViewSet  will be implemented in another issue